### ISSUE TITLE
Mouse clicks don't account for Windows 10 primary mouse button assignment

### ISSUE THREAD

**Issue by zilch42** [dictation-toolbox/Caster#719]
I mostly use a left-handed mouse because my main issue is with my right hand, and I just realised that Caster's mouse click rules don't account for Windows 10's primary mouse button settings. 

I.e. if your primary mouse button is set to "Right" in Window's settings, "Kick" gives you a right click rather than a left click and "Psychic" gives you the opposite as well. 

It would be great if those left and right click rules could be assigned to primary and secondary mouse clicks, rather than literal left and right mouse clicks.

**Comment by LexiconCode** [comment]
Good suggestion although I don't think it qualifies as a bug but an enhancement to Dragonfly's Mouse action. Perhaps Dragonfly could account for Windows settings for mouse mapping rather than then the static assignment of literal mouse mappings. Thoughts @Danesprite? otherwise I could work on something on this end.

Temporarily you could reverse the literal right and left assignments as a workaround. The mouse button assignments take place mostly in the nav, nav2  grammars using dragonfly [Mouse](https://dragonfly2.readthedocs.io/en/latest/actions.html#mouse-action) action.

**Comment by LexiconCode** [comment]
reference [SwapMouseButton function](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-swapmousebutton?redirectedfrom=MSDN)

**Comment by Danesprite** [comment]
@zilch42 Thanks for reporting this. Unfortunately, I am unable to reproduce the issue in my Windows 10 VM. The mouse buttons behave correct for me with the primary button set to left or right. Maybe this is a Windows bug?

Also, thanks for mentioning that function @LexiconCode :+1:. I wasn't aware it existed.

**Comment by Danesprite** [comment]
Actually, never mind what I said about not being able to reproduce this. I misunderstood what the problem was. I'm kind of surprised this hasn't come up before.

I think temporarily reversing the primary button with `SwapMouseButton(0)` and checking the return value should be good enough to fix this. If the result is `1`, then we need to swap back after using `SendInput` somewhere in Dragonfly's `actions.sendinput` or `actions.mouse._win32` modules. I can make a Dragonfly PR on this if you're not already working on it @LexiconCode?

**Comment by LexiconCode** [comment]
@Danesprite yeah I could see what I can do! Might be a bit busy even the next few days but we'll see.
